pandas/tests/io/excel/test_writers.py

```diff
@@ -1162,6 +1162,20 @@ def test_path_local_path(self, engine, ext):
                                        path="foo.{ext}".format(ext=ext))
         tm.assert_frame_equal(result, df)
 
+    def test_merged_cell_custom_objects(self, engine, ext):
+        # see GH-27006
+        mi = MultiIndex.from_tuples([(pd.Period('2018'), pd.Period('2018Q1')),
+                                     (pd.Period('2018'), pd.Period('2018Q2'))])
+        expected = DataFrame(np.random.rand(2, len(mi)), columns=mi)
+        expected.to_excel(self.path)
```

Comment:
Once you add the `merge_cells` fixture do `merge_cells=merge_cells` as part of the call here

Reply:
So if I do this, won't this always fail the assert comparison? When this is False, it creates a single cell that contains both levels in the excel file (for instance '2018.2018Q2'). When I read this back in, it's read in as a single column Index and fails comparing against the original MultiIndex.